plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -950,13 +950,27 @@ public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHan
     @Override
     public void createSchema(ConnectorSession session, String schemaName)
     {
-        execute(session, "CREATE SCHEMA " + quoted(schemaName));
+        JdbcIdentity identity = JdbcIdentity.from(session);
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            schemaName = toRemoteSchemaName(identity, connection, schemaName);
+            execute(connection, "CREATE SCHEMA " + quoted(schemaName));
```

Comment:
I would consider extracting `createSchemaSql` methods. Same for `dropSchemaSql`.